tests/testsuite/clean.rs

```diff
@@ -676,6 +676,20 @@ fn clean_spec_reserved() {
         .run();
 }
 
+#[cargo_test]
+fn clean_dry_run_gated() {
```

Comment:
I thought the other PR is making it stable, why is this commit within the PR undoing that?

Reply:
This was written before #12638, where it was decided to go ahead and stabilize the dry-run option. After rebasing, this should be resolved.